interface/modules/zend_modules/module/Installer/config/module.config.php

```diff
@@ -62,9 +62,7 @@
         'layout' => 'site/layout',
     ),
     'session' => array(
-                    'remember_me_seconds' => 2419200,
-                    'use_cookies' => true,
-                    'cookie_httponly' => true,
```

Comment:
@adunsulag ,
Since I am still Zend challenged, will this cause any security issues (on testing, it works well). Note the cookie_httponly was causing a very large bug (OpenEMR needs httponly off to support multiple logins by same user; allows users to open different patients simultaneously without sessions clashing) which became unmasked when always loaded it in the event dispatcher, so I had to remove that. And it just seemed sensical to remove the other settings since see no reason for zend to touch the cookie.

Reply:
@bradymiller I also wouldn't have expected Zend to deal with the cookie.  If it was creating a bug then somewhere zend is messing with the cookie and I need to find out where and for what reason. I can look into that.

On a related note, we have HTTP cookie's turned off everywhere?  I wasn't aware of that.  Are we just protecting ourselves than from XSS by using the CSRF?  HTTP only cookies prevent JS from reading the cookies and doing session hijacking.  I've been under the assumption all this time that the JS client can't read the cookies in OpenEMR.  That makes it more critical that we use a CSRF in the Reach work that matrix has done.

Reply:
Hi @adunsulag ,

That would be awesome if you could ensure this part of code is not introducing a security hole wherever it is used. It must be used to either to a session_start() or setcookie() which is overriding the settings of the original session cookie.

I just did some updates in the session/cookie that OpenEMR uses here and also added lots of documentation there:
https://github.com/openemr/openemr/blob/master/interface/globals.php#L92-L141

Here's the docs for the rationale of why OpenEMR needs cookie_httponly set to false (https://github.com/openemr/openemr/blob/master/interface/globals.php#L99-L104). If you don't set this to false, then it is actually a patient safety issue. Most web based apps can turn this to true, because they want a logged in user to use the same session/login even if they open separate tabs. However, OpenEMR, is different, since users open up different tabs in order to access different patients (thus the sessions/logins need to be separate). The clever mechanism to do this was created by @sunsetsystems (Rod) which requires being able to edit the session cookie (this is what all those top.restoreSession() calls do in the codebase) by using this code: https://github.com/openemr/openemr/blob/master/library/restoreSession.php#L20-L44

This is why OpenEMR needs to ensure no XSS vulnerabilities.